beekeeper-scheduler/src/main/java/com/expediagroup/beekeeper/scheduler/service/ExpiredHousekeepingMetadataSchedulerService.java

```diff
@@ -79,6 +83,61 @@ private HousekeepingMetadata createOrUpdateHousekeepingMetadata(HousekeepingMeta
     existingHousekeepingMetadata.setCleanupDelay(housekeepingMetadata.getCleanupDelay());
     existingHousekeepingMetadata.setClientId(housekeepingMetadata.getClientId());
 
+    if (isPartitionedTable(housekeepingMetadata)) {
+      compareMaxCleanupTimestampWithTable(existingHousekeepingMetadata);
+    }
+
     return existingHousekeepingMetadata;
   }
+
+  /**
+   * When the cleanup delay of a table with partitions is altered, the delay should be changed but the cleanup timestamp
+   * should be the max timestamp of any of the partitions which the table has.
+   *
+   * i.e. if the cleanup delay was 10 but now its being updated to 2, the delay should match any partition which has
+   * delay 10 (or above) to prevent premature attempts to cleanup the table.
+   *
+   * @param housekeepingMetadata
+   */
+  private void compareMaxCleanupTimestampWithTable(HousekeepingMetadata housekeepingMetadata) {
```

Comment:
To me "compare" suggests a method that just compares but doesn't have any side effects, while this and the one below both change something based on the result of the comparison. Can we rename this method to be clearer, maybe something like `updateCleanupTimestamp` or `setCleanupTimestampToMax`?

Reply:
Yeah you make a good point, will rename them 👍